enhancements/console/customize-add-page.md

```diff
@@ -0,0 +1,171 @@
+
+![Current Add page screenshot](customize-add-page.png)
+
+### Goals
+
+- Add configuration options to allow cluster admins to hide some entries from the add page.
+- Provide a list of available actions so that the admin can hide them.
+
+### Non-Goals
+
+- Introducing a UI to configure this without a yaml editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can hide actions from the add page.
+
+The cluster admin could disable actions via its id. This ensures that new later installed operators can show new actions without the requirement to enable them manually. To hide a complete group the admin should list all action ids of that group.
+
+A final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    addPage:
+      hideActions:
+      - devcatalog/databases
+      - devcatalog/helmcharts
+      - devcatalog/operatorbacked
+      - devcatalog/virtualmachinetemplates
+      - devcatalog/managedservices
+      - fromgit
+      - fromdevfile
+      - fromdockerfile
+      - containerimage
+      - importyaml
+      - uploadajar
+      - knative.openshift.io/eventsource
+      - knative.openshift.io/channel
+      - tekton.dev/pipelines
+  ...
+```
+
+### Provide a code snippet in YAML editor
+
+The admin needs to know the current available actions to hide them. We don't want to provide the defaults in the CR so that we can change these defaults later without migrating the customer data.
+
+To provide the default values close to the editor, the console will provide a code snippets in the sidebar of the YAML editor when editing the `Console` operator config. The snippets should include the ids of all extensions.
+
+### User Stories
+
+#### Story 1
+
+As an cluster admin, I want to customize the add page. The admin needs to know the current list of available actions. The admin can configure a list of actions which should not be shown on the add page.
+
+#### Story 2
+
+As a user, I don't want to see any actions on the add page which are hidden by the cluster admin. Empty groups should not be displayed.
+
+### Risks and Mitigations
+
+#### How we handle upgrade
+
+**Mitigation**: The cluster admin can provide a list of actions which should not shown to the users. All features which are not defined in that list will be shown after an upgrade.
+
+**Drawback** If the cluster admin installs a new OpenShift upgrade the customized add page will contain new or unexpected actions.
+
+## Open Questions
+
+- [x] Should we save a list of enabled actions instead of a list of ids which should not be shown?
+
+For example:
+
+```yaml
+spec:
+  customization:
+    addPage:
+      actions:
```

Comment:
```suggestion
      disabledActions:
```

Reply:
This is an example for a positive list. Its under the question (Should we save a list of enabled actions instead of a list of ids which should not be shown?). So actions or enabledActions makes is fine here.

Under this example there is an answer that we don't want go this way.